\documentclass{standalone}
\usepackage{standalone}
\documentclass[conference]{IEEEtran}
\bibliographystyle{ieeetran}
\bibliography{references}
%\begin{document}
%\begin{thebibliography}{99}
%\bibitem{KMC2}
%{\bf Deorowicz S., Kokot M., Grabowski S., Debudaj-Grabysz A.} (2015).
%KMC 2: Fast and resource-frugal k-mer counting. \emph{Bioinformatics}, doi: 10.1093/bioinformatics/btv022.
%
%\bibitem{KMC}
%{\bf Deorowicz S., Grabowski S., Debudaj-Grabysz A.} (2015).
%Disk-based k-mer counting on a PC. \emph{BMC Bioinformatics}, doi: 10.1186/1471-2105-14-160.
%
%\bibitem{MSPKmerCounter}
%{\bf Li Y., and Yan X.} (2014). MSPKmerCounter: A fast and
%memory efficient approach for k-mer counting. \emph{Preprint at
%http://cs.ucsb.edu/~yangli/papers/MSPKmerCounter.pdf}
%
%%\bibitem{Jellyfish}
%%{\bf Marcais G., and Kingsford C.} (2011). A fast, lock-free approach for efficient parallel
%%counting of occurrences of k-mers. \emph{Bioinformatics}, 27(6), 764?770.
%%
%%\bibitem{BFCounter}
%%{\bf Melsted P., and Pritchard J. K.} (2011). Efficient counting of k-mers in DNA sequences
%%using a Bloom Filter. \emph{BMC Bioinformatics}, 12(333).
%
%\bibitem{DSK}
%{\bf Rizk, G., Lavenier, D., and Chikhi R.} (2013). DSK: k-mer counting with very low
%memory usage. \emph{Bioinformatics}, 29(5), 652?653.
%
%%\bibitem{Turtle}
%%{\bf Roy, R. S., Bhattacharya D., and Schliep A.} (2014). Turtle: Identifying
%%frequent k-mers with cache-efficient algorithms. \emph{Bioinformatics},
%%doi:10.1093/bioinformatics/btu132.
%%
%%\bibitem{CeleraAssembler}
%%{\bf Miller J., Delcher A., Koren S., et al.} (2008). Aggressive assembly of pyrosequencing reads with mates. \emph{Bioinformatics},
%%doi: 10.1093/bioinformatics/btn548.
%%
%%\bibitem{Tallymer}
%%{\bf Kurtz S. et al.} (2008). A new method to compute k-mer frequencies and its application
%%to annotate large repetitive plant genomes. \emph{BMC Genomics},
%%9, 517.
%%
%%\bibitem{deBruijnGraph}
%%{\bf Compeau P., Pevzner P., Tesler G.} (2011). How to apply de Bruijn graphs to genome assembly. \emph{Nature Biotechnology},
%%29 (11): 987-991. 10.1038/nbt.2023.
%%
%%\bibitem{fastMultipleSeqAlign}
%%{\bf Edgar R.} (2004). MUSCLE: multiple sequence alignment with high accuracy and high throughput. \emph{Nucleic Acids Research},
%%32 (5): 1792-1797. 10.1093/nar/gkh340.
%%
%%\bibitem{KHmer}
%%{\bf Zhang Q., Pell J., Canino-Koning R., et al.} (2014). These are not the k-mers you are looking for: Efficient online k-mer counting using a probabilistic data structure. \emph{PLoS One},
%%9, e101271.
%%
%%\bibitem{countMinSketch}
%%{\bf Cormode, G., Muthukrishnan, S.} (2005).  An improved data stream summary: The count-min sketch and its applications.\emph{Journal of Algorithms},
%%55, 58???75.
%%
%%\bibitem{tandemRepeat}
%%{\bf Hani, Z.} (2015).  Red: an intelligent, rapid, accurate tool for detecting repeats de-novo on the genomic scale. \emph{BMC Bioinformatics},
%%doi: 10.1186/s12859-015-0654-5.
%%
%%\bibitem{reviewAllInOne}
%%{\bf Nelson P., Gutierrez M., Nelson V.} (April 4, 2016).  Computational Performance Assessment of k-mer Counting Algorithms. \emph{Journal of Computational Biology},
%%23(4): 248-255. doi:10.1089/cmb.2015.0199.
%%
%%\bibitem{KAnalyze}
%%{\bf Audano P., Vannberg F.} (2014).  KAnalyze: a fast versatile pipelined K-mer toolkit. \emph{Bioinformatics},
%%10.1093/bioinformatics/btu152.
%%
%%\bibitem{KCMBT}
%%{\bf Mamun A., Pal S., Rajasekaran S.} (2015).  Efficient techniques for k-mer counting \emph{IEEE Xplore Digital Library},
%%doi:10.1109/ICCABS.2015.7344733.
%
%\bibitem{paperOfKoren}
%{\bf Berlin K., Koren S. et al.} (2015).  Assembling large genomes with single-molecule sequencing and locality-sensitive hashing. \emph{Nature Biotechnology},
%33, 623???630 (2015) doi:10.1038/nbt.3238.
%%
%%\bibitem{bookOfCoreman}
%%{\bf Cormen T., et al.} (1990). Introduction to Algorithms. Chapter 12. \emph{MIT Press},
%%Cambridge,
%%MA.
%%
%%\bibitem{bookOfMinhash}
%%{\bf Rajaraman A., Ullman J.} (2011). Mining of Massive Datasets. Chapter 3. \emph{Cambridge University Press},
%%Cambridge,
%%MA.
%%
%%\bibitem{quake}
%%{\bf Kelley D., Schatz M., and Salzberg S.} (2010).  Quake: quality-aware detection
%%and correction of sequencing errors. \emph{Genome Biology},
%% 11(11):R116.
%%
%%\bibitem{readErrorCorrector1}
%%{\bf Li H.} (2015). Correcting Illumina sequencing errors for human data \emph{Cornell University Library},
%%arxiv.org/abs/1502.03744
%%
%%\bibitem{readErrorCorrector}
%%{\bf Deorowicz S., Dlugosz M.} (2016). RECKONER: Read Error Corrector Based on KMC \emph{Cornell University Library},
%%arxiv.org/abs/1602.03086
%
%\end{thebibliography}
%\end{document}